### DISCUSSION TITLE
Menu 22 ScnRev TIMEOUT

### DISCUSSION THREAD

**Discussion by OE1MWW** [egzumer/uv-k5-firmware-custom#222] [Ideas]
TIMEOUT - resumes scan after 5 seconds pause. Sometimes it takes a little bit longer for the reply of the other station.
Is there a way, or even a possible patch in the binary to extend that 5 seconds time - give me a hint please, in which
source this can be changed.

compiling, I got this error:

In file included from app/action.c:40:
c:/projects/uv-k5-firmware-custom/misc.h:24:24: error: expected declaration specifiers or '...' before '(' token
   24 |  #define ARRAY_SIZE(x) (sizeof(x) / sizeof(x[0]))
      |                        ^
app/action.c:99:15: note: in expansion of macro 'ARRAY_SIZE'
   99 | static_assert(ARRAY_SIZE(action_opt_table) == ACTION_OPT_LEN);
      |               ^~~~~~~~~~
make: *** [app/action.o] Error 1

**Comment by egzumer** [comment]
https://github.com/egzumer/uv-k5-firmware-custom/blob/main/app/chFrScanner.c#L98

I would use CARRIER if you are interested in what is being transmitted. TIMEOUT is just to give a hint on what is on the channel and move on.

Error fixed

**Reply by OE1MWW** [reply]
Thanks for the link - unfortunately I am fighting with this ARRAY:SIZE bug. Could not solve it.
Reason for more than 5 seonds: if there is a QSO on a channel you are running over in scan mode, it may happen, that the other QSO partner does not reply within 5 seconds. So, there was this wish of a Ham Radio buddy, to extend that time to wait for the other station.

**Reply by egzumer** [reply]
It seems that you are not asking about the time setting for TIMEOUT option then. TIMEOUT doesn't care if there is a transmission on a channel.

The error is fixed, pull the changes form the repository

**Reply by OE1MWW** [reply]
Thanks for the correction - no more bug warning ;-)

Î changed in misc.c

const uint16_t    scan_pause_delay_in_1_10ms       =  10000 / 10;   // 10 seconds

And I think that's what my friend was asking for. During scan, if there is a continous (conversation) on a chanel, the scan 
is holding the chanel for 10 seconds and continues after that . If the carrier is gone, for example after 3 or 5 seconds, the 
scan will wait 10 seconds until it continues. Both time spans are for a 'scanner' (person) sufficient, to either ignore (wait for 
10 seconds) or grab the HT, still on that channel, to interrupt scanning manually by PTT. 
Btw., for test purpose, I gave him a 8 and a 10 seconds variant.

**Reply by OE1MWW** [reply]
Wanted to change the hang time after a carrier is gone and to continue scan, (Menu 22 is set to 'CARRIER'):

Changed https://github.com/egzumer/uv-k5-firmware-custom/blob/main/misc.c#L57 the 5000 to 8000, 
checked with a stop watch  - if scan is halted by a carrier and the carrier goes 'off', it takes only 5 seconds
and not 8 seconds, as set - to continue the scan.

Additional changed in https://github.com/egzumer/uv-k5-firmware-custom/blob/main/app/chFrScanner.c#L98
the 'scan_pause_delay_in_1_10ms;' straight to a value of '8000;' - but still - the scan continues after ~ 5 seconds.

 Where is my misunderstanding ? Can someone please give me a clue ;-)

**Reply by egzumer** [reply]
You asked about TIMEOUT, I gave you the setting for that option, and you are changing that and not CARRIER.

For CARRIER

https://github.com/egzumer/uv-k5-firmware-custom/blob/13b6bb06b22f5c975c2f8295720b3203ba7d479a/app/app.c#L386
https://github.com/egzumer/uv-k5-firmware-custom/blob/13b6bb06b22f5c975c2f8295720b3203ba7d479a/misc.c#L63

**Reply by OE1MWW** [reply]
sorry for my mistake - and many Thanks for your help ;-)
